numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -1966,13 +1968,15 @@ array_empty(PyObject *NPY_UNUSED(ignored),
     NPY_ORDER order = NPY_CORDER;
     npy_bool is_f_order;
     PyArrayObject *ret = NULL;
+    NPY_DEVICE device = NPY_DEVICE_CPU;
     PyObject *like = Py_None;
     NPY_PREPARE_ARGPARSER;
 
     if (npy_parse_arguments("empty", args, len_args, kwnames,
             "shape", &PyArray_IntpConverter, &shape,
             "|dtype", &PyArray_DTypeOrDescrConverterOptional, &dt_info,
             "|order", &PyArray_OrderConverter, &order,
+            "$device", &PyArray_DeviceConverterOptional, &device,
```

Comment:
I'd again argue to flip the order with `like`.

Reply:
Commented above about the `device, like` order.